### PR TITLE
usbmuxd: Rebuild against libplist 2.3.0

### PR DESCRIPTION
**Pull request by Cyanoxygen**
Topic Description
-----------------

- usbmuxd: update to 1.1.1+git20240916
    - Upstream have not published tags since June 2020.
    - Code in master brings support for more devices.
    - Also removed some redundant code.
    - These commits can not be individually picked.
- libplist: update to 2.6.0
    - Clean up build scripts.
    - Disable build for Python 2.
- libimobiledevice-glue: new, 1.3.1
    This is a library shared across all projects within
    libimobiledevice.org.

Package(s) Affected
-------------------

- libimobiledevice-glue: 1.3.1
- libplist: 2.6.0
- usbmuxd: 1.1.1+git20240916

Security Update?
----------------

No

Build Order
-----------

```
#buildit libplist libimobiledevice-glue usbmuxd
```

Test Build(s) Done
------------------

**Primary Architectures**

- [ ] AMD64 `amd64`
- [ ] AArch64 `arm64`
- [ ] LoongArch 64-bit `loongarch64`

**Secondary Architectures**

- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

### CODE REVIEW

**Comment by Cyanoxygen** [comment]
Testing
--------

LGTM on amd64, my iPhone can now share its network connection to my PC:

image

File: runtime-common/libplist/autobuild/beyond [hunk 1]
```diff
@@ -1,3 +1,9 @@
+if ab_match_archgroup mainline ; then
+	abinfo "Installing Cython declarations ..."
+	install -Dm644 "$SRCDIR"/cython/plist.pxd \
```
Comment (MingcongBai) [inline runtime-common/libplist/autobuild/beyond:3]: Verbosity here.

File: runtime-common/libplist/spec [hunk 2]
```diff
@@ -1,5 +1,6 @@
-VER=2.3.0
-REL=2
-SRCS="tbl::https://github.com/libimobiledevice/libplist/releases/download/$VER/libplist-$VER.tar.bz2"
-CHKSUMS="sha256::4e8580d3f39d3dfa13cefab1a13f39ea85c4b0202e9305c5c8f63818182cac61"
+UPSTREAM_VER=2.6.0
```
Comment (MingcongBai) [inline runtime-common/libplist/spec]: Unnecessary duplication.

File: runtime-devices/libimobiledevice-glue/spec [hunk 3]
```diff
@@ -0,0 +1,6 @@
+UPSTREAM_VER=1.3.1
+VER="$UPSTREAM_VER"
```
Comment (MingcongBai) [inline runtime-devices/libimobiledevice-glue/spec]: Unnecessary duplication.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

**Review by MingcongBai — APPROVED** [review]

**Pull request merged**